tests/test_distributed.py

```diff
@@ -0,0 +1,121 @@
+# Copyright 2022 MosaicML Streaming authors
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import math
+import os
+from typing import Tuple
+
+import pytest
+import torch.distributed as dist
+from torch.utils.data import DataLoader
+
+import streaming.base.distributed as ms_dist
+from streaming.base import Dataset
+
+from .common.datasets import *
+from .common.distributed import DistributedTest
+from .common.utils import *
+
+logger = logging.getLogger(__name__)
+
+
+class TestWorldSize(DistributedTest):
+    world_size = 3
+
+    def test_class_variable(self):
+        assert dist.is_initialized()
+        assert dist.get_world_size() == 3
+        assert dist.get_rank() < 3
+
+    @pytest.mark.world_size(4)
+    def test_parameterize(self):
+        assert dist.is_initialized()
+        assert dist.get_world_size() == 4
+        assert dist.get_rank() < 4
+
+
+# @pytest.mark.skip
+class TestInit(DistributedTest):
+    world_size = 2
+
+    @pytest.mark.parametrize('batch_size', [128])
+    @pytest.mark.parametrize('drop_last', [False, True])
+    @pytest.mark.parametrize('num_workers', [1, 8])
```

Comment:
Should we add zero here?

...('num_workers', [0, 1, 8])?

Reply:
Added in https://github.com/mosaicml/streaming/pull/57/commits/8741970777041113ee42870ed439f33131211835.